package.json

```diff
@@ -22,8 +22,10 @@
     "test": "mocha --timeout 10000",
     "lint": "npm run lint:js",
     "lint:fix": "npm run lint:fix:js",
-    "lint:js": "eslint .",
-    "lint:fix:js": "eslint --fix ."
+    "lint:js": "npm run lint:js:eslint && npm run lint:js:types",
+    "lint:fix:js": "eslint --fix .",
+    "lint:js:eslint": "eslint .",
+    "lint:js:types": "!(npx tsc --allowJs --noEmit $(cat jsconfig.json | jq -r '.compilerOptions | to_entries | map([\"--\\(.key)\",.value]) | flatten | join(\" \")') src/**.js | grep \"^\\(src\\|bin\\|test\\|examples\\)/\")"
```

Comment:
This nasty dance lets us do two things:
- Pull most of the compiler settings from `jsconfig.json`.
- Only look at errors from our files (there's no way to get the type checking we want and have TypeScript only report errors in our files---`node_modules` either reports errors or the type information from modules isn't used for our files).
- The `!` up front inverts the exit code from `grep`. `grep` returns `0` when it finds a match and `1` when it finds no match, but we want the command to *fail* when `grep` finds a match and *pass* when it finds no match, so we invert the exit code.